File: scan-gacor.py
import os
import sys
from datetime import datetime

# Mencoba membuka file wordlist
keywords = None
try:
    with open("wordlist.txt", "r") as wordlist:
        # Ubah menjadi array
        keywords = wordlist.read().splitlines()
        print("Menggunakan file wordlist.")
except FileNotFoundError:
    print("File wordlist tidak ditemukan.")
    print("Menggunakan wordlist default.")

    # Daftar kata kunci yang biasanya terkait dengan situs judi
    keywords = ['judi', 'poker', 'kasino', 'slot', 'taruhan', 'togel', 'mahjong']

# Fungsi untuk memindai file
def scan_file(file_path, keywords):
    with open(file_path, "r", encoding="utf-8", errors="ignore") as file:
        content = file.read()
        found = []
        for keyword in keywords:
            if keyword.lower() in content.lower():
                print(f'Kata kunci "{keyword}" ditemukan dalam file {file_path}')
                found.append(keyword)
        return found

# Fungsi untuk memindai folder
def scan_folder(folder_path, keywords):
    found = []
    for keyword in keywords:
        if keyword.lower() in folder_path.lower():
            print(f'Kata kunci "{keyword}" ditemukan dalam folder {folder_path}')
            found.append(keyword)
    return found

# Fungsi untuk memindai direktori
def scan_directory(directory_path, keywords):
    results = []
    for root, dirs, files in os.walk(directory_path):
        for file in files:
            if file.lower().endswith(('.php', '.html')):
                file_path = os.path.join(root, file)
                r = scan_file(file_path, keywords)
                if r:
                    results.append({
                        "type": "file",
                        "found": ",".join(r),
                        "path": file_path
                    })
        for folder in dirs:
            folder_path = os.path.join(root, folder)
            r = scan_folder(folder_path, keywords)
            if r:
                results.append({
                    "type": "folder",
                    "found": ",".join(r),
                    "path": folder_path
                })
    return results

# Mendapatkan argumen direktori dari command line
directory_path = sys.argv[1] if len(sys.argv) > 1 else ''

# Cek argumen -d untuk delete file yang terdeteksi
delete = '-d' in sys.argv

# Cek argumen -l untuk membuat file log
log = '-l' in sys.argv

if directory_path and os.path.isdir(directory_path):
    print("Memulai scan...")
    rdir = scan_directory(directory_path, keywords)
    if delete:
        print("Menghapus file dan folder yang terdeteksi...")
        for r in rdir:
            if r['type'] == 'file':
                os.remove(r['path'])
            else:
                os.rmdir(r['path'])
    if log:
        # Membuat file log dengan new line
        # filename dengan format log-<datetime>.txt
        filename = f"log-{datetime.now().strftime('%Y-%m-%d-%H-%M-%S')}.txt"
        with open(filename, "w") as log_file:
            for r in rdir:
                log_file.write(f"{r['path']} | keyword: {r['found']}\n")
    print("Selesai scan.")
else:
    print("Argumen direktori tidak valid atau tidak diberikan.")

File: scan-gacor.php
<?php
//mencoba membuka file wordlist
$keywords = [];
try {
    $file = fopen("wordlist.txt", "r");
    while (!feof($file)) {
        $line = fgets($file);
        $line = trim($line);
        if ($line !== '') {
            $keywords[] = $line;
        }
    }

    fclose($file);
} catch (Exception $e) {
    echo "File wordlist tidak ditemukan.\n";
    echo "Menggunakan wordlist default.\n";

    // Daftar kata kunci yang biasanya terkait dengan situs judi
    $keywords = ['judi', 'poker', 'kasino', 'slot', 'taruhan', 'togel', 'mahjong'];
}



// Fungsi untuk memindai file
function scan_file($file_path, $keywords)
{
    $content = file_get_contents($file_path);
    $found = [];
    foreach ($keywords as $keyword) {
        if (stripos(strtolower($content), strtolower($keyword)) !== false) {
            echo 'Kata kunci "' . $keyword . '" ditemukan dalam file ' . $file_path . "\n";
            $found[] = $keyword;
        }
    }
    return $found;
}

function scan_folder($folder_path, $keywords)
{
    $found = [];
    foreach ($keywords as $keyword) {
        if (stripos(strtolower($folder_path), strtolower($keyword)) !== false) {
            echo 'Kata kunci "' . $keyword . '" ditemukan dalam folder ' . $folder_path . "\n";
            $found[] = $keyword;
        }
    }
    return $found;
}


// Fungsi untuk memindai direktori
function scan_directory($directory_path, $keywords)
{
    $files = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($directory_path));
    $results = [];
    foreach ($files as $file) {
        if ($file->isFile() && in_array($file->getExtension(), ['php', 'html'])) {
            $r = scan_file($file->getRealPath(), $keywords);
            if (count($r) > 0) {
                $results[] = [
                    "type" => "file",
                    "found" => implode(",", $r),
                    "path" => $file->getRealPath()
                ];
            }
        } else if ($file->isDir()) {
            $r = scan_folder($file->getRealPath(), $keywords);
            if (count($r) > 0) {
                $results[] = [
                    "type" => "folder",
                    "found" => implode(",", $r),
                    "path" => $file->getRealPath()
                ];
            }
        }
    }
    return $results;
}

// Mendapatkan argumen direktori dari command line
$directory_path = $argv[1] ?? '';

//cek argumen -d untuk delete file yang terdeteksi
$delete = false;
$log = false;
foreach ($argv as $arg) {
    if ($arg == '-d') {
        $delete = true;
    }
    if ($arg == '-l') {
        $log = true;
    }
}


if ($directory_path !== '' && is_dir($directory_path)) {
    echo "Memulai scan...";
    $rdir = scan_directory($directory_path, $keywords);
    if ($delete) {
        echo "Menghapus file dan folder yang terdeteksi...";
        for ($i = 0; $i < count($rdir); $i++) {
            if ($rdir[$i]['type'] == 'file') {
                unlink($rdir[$i]['path']);
            } else {
                rmdir($rdir[$i]['path']);
            }
        }
    }
    if ($log) {
        //membuat file log dengan new line
        $log = fopen("log-".date("Y-m-d-H-i-s").".txt", "w");
        foreach ($rdir as $r) {
            fwrite($log, $r['path'] . " | keyword : " . $r['found'] . "\n");
        }
        fclose($log);
    }
    echo "Selesai scan.";
} else {
    echo "Argumen direktori tidak valid atau tidak diberikan.\n";
}
?>